pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -20,24 +20,482 @@
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceToffoli(qml.Toffoli, re.ResourceOperator):
+    r"""Resource class for Toffoli
+
+    Resources:
+        The resources are obtained from (in figure 1.) the paper `Novel constructions for the fault-tolerant
+        Toffoli gate <https://arxiv.org/pdf/1212.5069>`_.
+
+        The circuit which applies the Toffoli gate on target wire 'target' with control wires ('c1', 'c2') is
+        given by:
+
+        .. code-block:: bash
+
+                c1: ─╭●────╭X──T†────────╭X────╭●───────────────╭●─┤
+                c2: ─│──╭X─│──╭●───T†─╭●─│──╭X─│────────────────╰Z─┤
+              aux1: ─╰X─│──│──╰X───T──╰X─│──│──╰X────────────────║─┤
+              aux2: ──H─╰●─╰●──T─────────╰●─╰●──H──S─╭●──H──┤↗├──║─┤
+            target: ─────────────────────────────────╰X──────║───║─┤
+                                                             ╚═══╝
+
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        gate_types = {}
+
+        cnot = re.ResourceCNOT.resource_rep()
+        t = re.ResourceT.resource_rep()
+        h = re.ResourceHadamard.resource_rep()
+        s = re.ResourceS.resource_rep()
+        cz = re.ResourceCZ.resource_rep()
+
+        gate_types[cnot] = 9
+        gate_types[h] = 3
+        gate_types[s] = 1
+        gate_types[cz] = 1
+        gate_types[t] = 2 + 2 * (7)  # T^dagg = 7*T in cost  TODO: Update with Adjoint(T)
+
+        return gate_types
+
+    @staticmethod
+    def textbook_resource_decomp() -> Dict[re.CompressedResourceOp, int]:
+        r"""Resources for the Toffoli gate
 
         Resources:
-            The resources come from the following identity expressing Controlled Phase Shift
-            as a product of Phase Shifts and CNOTs.
+            The resources are taken (figure 4.9) from the textbook `Quantum Computation and Quantum Information
+            <https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE#overview>`_.
+
+            The circuit is given by:
+
+            .. code-block:: bash
+
+                0: ───────────╭●───────────╭●────╭●──T──╭●─┤
+                1: ────╭●─────│─────╭●─────│───T─╰X──T†─╰X─┤
+                2: ──H─╰X──T†─╰X──T─╰X──T†─╰X──T──H────────┤
+
+        """
+        gate_types = {}
+
+        cnot = re.ResourceCNOT.resource_rep()
+        t = re.ResourceT.resource_rep()
+        h = re.ResourceHadamard.resource_rep()
+
+        gate_types[cnot] = 6
+        gate_types[h] = 2
+        gate_types[t] = 7
+
+        return gate_types
+
+    def resource_params(self) -> dict:
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceMultiControlledX(qml.MultiControlledX, re.ResourceOperator):
+    r"""Resource class for MultiControlledX
```

Comment:
```suggestion
    r"""Resource class for the MultiControlledX gate.
```
For consistency (see `ops/qubit/parametric_ops_single_qubit.py`.
Likewise for the other controlled ops.